pkgs/tools/networking/ddclient/default.nix

```diff
@@ -25,7 +26,12 @@ buildPerlPackage rec {
   '';
 
   installPhase = ''
+    runHook preInstall
```

Comment:
Perhaps it's better to change installPhase to postInstall, so the hooks don't have to be invoked manually?

Reply:
I like how this is more explicit - everything that happens for the install happens here.

Is there a "recommended pattern"?

Reply:
I don't know if it's written down, but I think it's recommended to not override the phases directly, but rather add stuff to pre/post. It's easy to forget to invoke the hooks. (As evidenced by this expression that you're fixing now.)

Reply:
The thing is, this *is* the installation phase. From the perspective of "least surprising behaviour" I would assume installation is done in the ```installPhase```. If we do this in ```postInstall``` instead, somebody cannot override that phase without not having anything installed. 

I fully understand where you're coming from though - when people start messing with ```phases``` directly there is a high chance of someone breaking things or accidentally leaving out phases that should be run. But in this case this is actually doing the right thing (IMHO).

Reply:
Ok.